tests/tests_pytorch/strategies/test_colossalai_strategy.py

```diff
@@ -0,0 +1,269 @@
+
+        self.train_acc = Accuracy()
+        self.valid_acc = Accuracy()
+        self.test_acc = Accuracy()
+
+    def build_layers(self) -> nn.Module:
+        layers = []
+        for _ in range(3):
+            layers.append(nn.Linear(32, 32))
+            layers.append(nn.ReLU())
+        layers.append(nn.Linear(32, 3))
+        return nn.Sequential(*layers)
+
+    def configure_sharded_model(self) -> None:
+        if self.layers is None:
+            self.layers = self.build_layers()
+
+    def forward(self, x):
+        x = self.layers(x)
+        logits = F.softmax(x, dim=1)
+        return logits
+
+    def configure_optimizers(self):
+        optimizer = HybridAdam(self.parameters(), lr=self.lr)
+        return [optimizer], []
+
+    def training_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self.forward(x)
+        loss = F.cross_entropy(logits, y)
+        self.log("train_loss", loss, prog_bar=True, sync_dist=True)
+        self.log("train_acc", self.train_acc(logits, y), prog_bar=True, sync_dist=True)
+        return {"loss": loss}
+
+    def validation_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self.forward(x)
+        self.log("val_loss", F.cross_entropy(logits, y), prog_bar=False, sync_dist=True)
+        self.log("val_acc", self.valid_acc(logits, y), prog_bar=True, sync_dist=True)
+
+    def test_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self.forward(x)
+        self.log("test_loss", F.cross_entropy(logits, y), prog_bar=False, sync_dist=True)
+        self.log("test_acc", self.test_acc(logits, y), prog_bar=True, sync_dist=True)
+
+    def predict_step(self, batch, batch_idx):
+        x, _ = batch
+        return self.forward(x)
+
+
+@RunIf(min_cuda_gpus=2, standalone=True, colossalai=True)
+def test_multi_gpu_model_colossalai_fit_only(tmpdir):
+    dm = ClassifDataModule()
+    model = ModelParallelClassificationModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir, accelerator="gpu", devices=2, precision=16, strategy="colossalai", max_epochs=1
+    )
+    trainer.fit(model, datamodule=dm)
+
+
+@RunIf(min_cuda_gpus=2, standalone=True, colossalai=True)
+def test_multi_gpu_model_colossalai_test_only(tmpdir):
+    dm = ClassifDataModule()
+    model = ModelParallelClassificationModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir, accelerator="gpu", devices=2, precision=16, strategy="colossalai", max_epochs=1
+    )
+    trainer.test(model, datamodule=dm)
+
+
+@RunIf(min_cuda_gpus=2, standalone=True, colossalai=True)
+def test_multi_gpu_model_colossalai_fit_test(tmpdir):
+    seed_everything(4321)
+    dm = ClassifDataModule()
+    model = ModelParallelClassificationModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        accelerator="gpu",
+        devices=2,
+        precision=16,
+        strategy=ColossalAIStrategy(initial_scale=32),
+        max_epochs=1,
+    )
+    trainer.fit(model, datamodule=dm)
```

Comment:
can you assert train and val accuracy? also is it possible to limit the batches here, keeping the accuracy benchmark the same?

Reply:
ok.